app/models/role.rb

```diff
@@ -75,6 +75,27 @@ class Role < ApplicationRecord
   scoped_search :on => :builtin, :complete_value => { :true => true, :false => false }
   scoped_search :on => :description, :complete_value => false
 
+  class << self
+    attr_accessor :bypass_locking
```

Comment:
I agree it makes sense to have a common name for both the class and instance version of this, as well as a wrapper function that ensures it returns to the original value.